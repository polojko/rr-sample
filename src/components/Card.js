import React from 'react';

const Card = ({card}) => {
    return (
        <div key={card.id} className='card'>
            <img src={card.imageUrl} alt={card.name} />
            <div className='card-name'>{card.name}</div>
            <div className='card-text'>{card.text}</div>
            <div className='other-fields'>
                {card.set.name && <span><b>Set:</b> {card.set.name} </span>}
                {card.type && <span><b>Type:</b> {card.type}</span>}
            </div>
        </div>
    )
};

export default Card;