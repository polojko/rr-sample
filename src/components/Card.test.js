import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

it('Card component renders correctly', () => {
    const cardData = {
        "name": "Raise Dead",
        "rarity": "Legendary",
        "type": "Action",
        "cost": 2,
        "set": {
            "id": "cs",
            "name": "Core Set",
            "_self": "https://api.elderscrollslegends.io/v1/sets/cs"
        },
        "collectible": false,
        "text": "Summon a random creature from each discard pile.",
        "attributes": [
            "Endurance"
        ],
        "unique": false,
        "imageUrl": "https://images.elderscrollslegends.io/cs/raise_dead.png",
        "id": "ce7be2e72d6b06a52e50bed01952801ca4ecfade"
    }
    const card = renderer
        .create(<Card card={cardData} />)
        .toJSON();
    expect(card).toMatchSnapshot();
    expect(cardData).toMatchSnapshot({
        name: 'Raise Dead',
        type: 'Action',
        set: {
            name: 'Core Set'
        },
        text: 'Summon a random creature from each discard pile.',
        imageUrl: 'https://images.elderscrollslegends.io/cs/raise_dead.png'
    });
});