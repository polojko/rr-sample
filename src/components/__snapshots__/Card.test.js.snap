// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="loader"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <p>
      Loading...
    </p>
  </div>
  <div
    className="card-name"
  >
    Raise Dead
  </div>
  <div
    className="card-text"
  >
    Summon a random creature from each discard pile.
  </div>
  <div
    className="other-fields"
  >
    <span>
      <b>
        Set:
      </b>
       
      Core Set
       
    </span>
    <span>
      <b>
        Type:
      </b>
       
      Action
    </span>
  </div>
  <div
    className="select-card"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
    >
      <span
        className="MuiSwitch-root Component-root-1 MuiSwitch-sizeSmall"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiSwitch-switchBase Component-switchBase-2 MuiSwitch-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-10 MuiSwitch-input"
              name="ce7be2e72d6b06a52e50bed01952801ca4ecfade"
              onChange={[Function]}
              type="checkbox"
              value=""
            />
            <span
              className="MuiSwitch-thumb Component-thumb-3"
            />
          </span>
        </span>
        <span
          className="MuiSwitch-track Component-track-4"
        />
      </span>
      <span
        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Select card:
      </span>
    </label>
  </div>
</div>
`;

exports[`Card component renders correctly 2`] = `
Object {
  "attributes": Array [
    "Endurance",
  ],
  "collectible": false,
  "cost": 2,
  "id": "ce7be2e72d6b06a52e50bed01952801ca4ecfade",
  "imageUrl": "https://images.elderscrollslegends.io/cs/raise_dead.png",
  "name": "Raise Dead",
  "rarity": "Legendary",
  "set": Object {
    "_self": "https://api.elderscrollslegends.io/v1/sets/cs",
    "id": "cs",
    "name": "Core Set",
  },
  "text": "Summon a random creature from each discard pile.",
  "type": "Action",
  "unique": false,
}
`;
